core/classifiers/classifier_utils.py

```diff
@@ -60,15 +63,80 @@ def extract_svm_parameters(clf):
     }
 
 
-def unicode_validator_for_classifier_data(var):
-    """Validates that incoming object contains unicode literal strings."""
-    if isinstance(var, dict):
-        for k in var.keys():
+def unicode_validator_for_classifier_data(classifier_data):
+    """Validates that incoming object contains unicode literal strings.
+
+    Args:
+        classifier_data: *. The trained classifier model data.
+
+    Raises:
+        Exception. If any of the strings in classifier data is not a unicode
+            string.
+    """
+    if isinstance(classifier_data, dict):
+        for k in classifier_data:
             if isinstance(k, str):
                 raise Exception('Expected %s to be unicode but found str.' % k)
-            unicode_validator_for_classifier_data(var[k])
-    if isinstance(var, (list, set, tuple)):
-        for item in var:
+            unicode_validator_for_classifier_data(classifier_data[k])
+    elif isinstance(classifier_data, list):
+        for item in classifier_data:
             unicode_validator_for_classifier_data(item)
-    if isinstance(var, str):
-        raise Exception('Expected \'%s\' to be unicode but found str.' % var)
+    elif isinstance(classifier_data, str):
+        raise Exception(
+            'Expected \'%s\' to be unicode but found str.' % classifier_data)
+    else:
+        return
+
+
+def convert_float_numbers_to_string_in_classifier_data(classifier_data):
+    """Converts all floating point numbers in classifier data to string.
+
+    The following function iterates through entire classifier data and converts
+    all float values to corresponding string values. At the same time, it also
+    verifies that none of the existing string values are convertible to float
+    values with the help of regex.
+
+    Args:
+        classifier_data: dict|list|string|int|float. The original classifier
+            data which needs conversion of floats to strings.
+
+    Raises:
+        Exception. If any of the string values are convertible to float then
+            an exception is raised to report the error. The classifier data
+            must not include any string values which can be casted to float.
+        Exception. If classifier data contains an object whose type is other
+            than integer, string, dict, floats or list.
+
+    Returns:
+        dict|list|string|int|float. Modified classifier data in which float
+            values are converted into strings.
+    """
+    if isinstance(classifier_data, dict):
+        for k in classifier_data:
```

Comment:
Make new return value called classifier_data_with_stringified_floats and return that, so that we're always clear when stuff has been processed versus not. Just calling everything classifier_data is a bit confusing. (I made a similar comment in the other PR FYI)